services/user-service/entry-points/api/routes.ts

```diff
@@ -0,0 +1,71 @@
+import express from 'express';
+import * as userUseCase from '../../domain/user-use-case';
+import { validateAddUserInput, validateUpdateUserInput } from './validators';
+
+export default function defineRoutes(expressApp: express.Application) {
+  const router = express.Router();
+
+  router.post('/',validateAddUserInput ,async (req, res, next) => {
+    try {
+      // ✅ Best Practice: Using the 3-tier architecture, routes/controller are kept thin, logic is encapsulated in a dedicated domain folder
+      const addUserResponse = await userUseCase.addUser(req.body);
+      return res.json(addUserResponse);
+    } catch (error) {
+      next(error);
+    }
+  });
+
+  router.get('/', async (req, res, next) => {
+    try {
+      const {email} = req.query
+      var response = await userUseCase.getAllUser();
+      if (email) {
+        response = response.filter(res => res.email === email)
+      }
+      res.json(response.length == 0? null : response);
+    } catch (error) {
+      next(error);
+    }
+  })
+
+  // get existing user by id
+  router.get('/:id', async (req, res, next) => {
+    try {
+      const response = await userUseCase.getUserById(parseInt(req.params.id, 10));
+      res.status(200).json(response);
+    } catch (error) {
+      next(error);
+    }
+  });
+
+  router.get('/email/:email', async (req, res, next) => {
+    try {
+      const response = await userUseCase.getUserByMail(req.params.email);
+      res.status(202).json(response);
+    } catch (error) {
+      next(error)
+    }
+  });
```

Comment:
For queries, please modify in the next PR. See: https://github.com/CS3219-AY2324S1/ay2324s1-assignment-6-g38/pull/7#discussion_r1349442975

Reply:
Noted sir!